docs/puzzles/singletons.md

```diff
@@ -0,0 +1,437 @@
+
+This puzzle is an outer puzzle and is used to wrap pooling puzzles, NFTs, and decentralized identities.  Any inner puzzle can be wrapped with this puzzle if it has necessity for uniqueness.
+
+## Design choices
+
+A few design choices were made in the creation of this puzzle so let's go over them now:
+
+* **Singletons are always odd.**  In order to assure that a singleton does not duplicate itself, it needs some way to verify that it's children do not consist of more than one new singleton.  It does this by verifying that only one of its children is odd.  It is either a new singleton, or it is the **melt value** and will be ignored (more on that later).  Odd was chosen over even because you may want to have singletons create other, non-singleton coins at times (like 0 amount messages for the DID) and having the singleton be odd just makes this easier.  Coins can be multiples of 10 so you can send a full XCH rather than an XCH and a mojo.  No matter how many even amounts you subtract from an odd amount, the end result will always be odd.
+* **Singletons always wrap their odd child.** This abstracts some of the singleton functionality away from the inner puzzles.  If an inner puzzle creates an odd coin, it doesn't have to worry about making it a singleton, the [outer puzzle will take care of that](https://chialisp.com/docs/common_functions/#outer-and-inner-puzzles).  It also prevents an inner puzzle from accidentally melting the singleton by forgetting to wrap its odd output.
+* **A specific melt value determines whether the singleton wraps its child.** If you would like to destroy a singleton and use its amount to create a new non-singleton coin, you need to output a `CREATE_COIN` condition that uses the amount `-113`.  When the singleton outer puzzle sees that condition, it filters it out.  This amount was arbitrarily chosen and is negative because a coin with negative amount cannot exist. Therefore, it is an unlikely accidental output of an inner puzzle.  This is chosen to prevent an inner puzzle from accidentally melting a singleton by forgetting to create an odd output.  It must deliberately specify to melt the singleton.  Keep in mind that the melt value *does* count as the odd output, but is ignored, creating the need to burn one mojo of coin value in order to melt (all of the outputs must be even).  Usually there will be a transaction fee anyways so it likely just becomes part of that.  You should also be wary of the amount of control you give to people who you partially lend your singleton to.  If they can freely create conditions, it may be possible for them to melt your singleton against your wishes.
```

Comment:
The canonical hacker term here is "magic" (as in "A specific magic melt value...").